fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1351,27 +1355,44 @@ private QueryComponent normalizeAndOrForInAsOr(@Nonnull QueryComponent component
     }
 
     @Nonnull
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
     private RecordQueryPlan tryToConvertToCoveringPlan(@Nonnull PlanContext planContext, @Nonnull RecordQueryPlan chosenPlan) {
         if (chosenPlan instanceof RecordQueryPlanWithIndex) {
             // Check if the index scan covers, then convert it to a covering plan.
-            return tryToConvertToCoveringPlan(planContext, (RecordQueryPlanWithIndex) chosenPlan);
+            return tryToConvertToCoveringPlan(planContext, (RecordQueryPlanWithIndex) chosenPlan, null);
         } else if (chosenPlan instanceof RecordQueryUnorderedPrimaryKeyDistinctPlan) {
             // If possible, push down the covering index transformation so that
             // it happens before checking for distinct primary keys
             final RecordQueryUnorderedPrimaryKeyDistinctPlan distinctPlan = (RecordQueryUnorderedPrimaryKeyDistinctPlan) chosenPlan;
             if (distinctPlan.getChild() instanceof RecordQueryPlanWithIndex) {
-                final RecordQueryPlan newChildPlan = tryToConvertToCoveringPlan(planContext, (RecordQueryPlanWithIndex) distinctPlan.getChild());
+                final RecordQueryPlan newChildPlan = tryToConvertToCoveringPlan(planContext, (RecordQueryPlanWithIndex) distinctPlan.getChild(), null);
                 if (newChildPlan != distinctPlan.getChild()) {
                     return new RecordQueryUnorderedPrimaryKeyDistinctPlan(newChildPlan);
                 }
             }
+        } else if (chosenPlan instanceof RecordQueryFilterPlan) {
+            final RecordQueryFilterPlan filterPlan = (RecordQueryFilterPlan)chosenPlan;
+            if (filterPlan.getChild() instanceof RecordQueryPlanWithIndex) {
+                final RecordQueryPlanWithIndex filteredIndexPlan = (RecordQueryPlanWithIndex) filterPlan.getChild();
+                final Index index = metaData.getIndex(filteredIndexPlan.getIndexName());
+                if (indexTypes.getValueTypes().contains(index.getType())) {
```

Comment:
A text index, in particular, does not have the actual indexed text in the entry.

Reply:
This is a great point. I think it's worth adding it as a comment here in case someone forgets this in the future.

Reply:
Added there.